mock/py/mockbuild/plugins/procenv.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# vim:expandtab:autoindent:tabstop=4:shiftwidth=4:filetype=python:textwidth=0:
+# License: GPL2 or later see COPYING
+
+# python library imports
+import codecs
+
+# our imports
+from mockbuild.trace_decorator import getLog, traceLog
+import mockbuild.util
+
+requires_api_version = "1.1"
+
+
+# plugin entry point
+@traceLog()
+def init(plugins, conf, buildroot):
+    ProcEnv(plugins, conf, buildroot)
+
+
+class ProcEnv(object):
+    # pylint: disable=too-few-public-methods
+    """Get the runtime process environment"""
+    @traceLog()
+    def __init__(self, plugins, conf, buildroot):
+        self.buildroot = buildroot
+        self.hw_info_opts = conf
+        self.config = buildroot.config
+        plugins.add_hook("prebuild", self._PreBuildHook)
+
+    # =============
+    # 'Private' API
+    # =============
+    @traceLog()
+    def _PreBuildHook(self):
+        getLog().info("enabled ProcEnv plugin")
+
+        self.buildroot.pkg_manager.install('procenv')
+
+        out_file = self.buildroot.resultdir + '/procenv.log'
+        with codecs.open(out_file, 'w', 'utf-8', 'replace') as out:
+
+            cmd = ["/usr/bin/procenv"]
```

Comment:
This add new dependence on package `procenv`. It needs to be added either as Requires or Suggests/Recommends

Reply:
Good call, I'll get that fixed.